src/library_int53.js

```diff
@@ -103,13 +117,22 @@ mergeInto(LibraryManager.library, {
     // convertU32PairToI53())
     assert(hi === (hi|0));
 #endif
-    return (lo >>> 0) + hi * 4294967296;
+    var result = (lo >>> 0) + hi * 4294967296;
+#if ASSERTIONS
+    if ((result >>> 0) != (lo >>> 0)) warnOnce('convertI32PairToI53() resulted in rounding lo=0x' + lo.toString(16) + ', hi=0x' + hi.toString(16) + ' to 0x' + result.toString(16));
+#endif
+    return result;
   },
 
-  // Converts the given unsigned 32-bit low-high pair to a JavaScript Number that can
-  // represent 53 bits of precision.
+  // Converts the given 32-bit low-high pair to an unsigned JavaScript number value
+  // (with rounding beyond 2^53).
+  // The signedness of each word doesn't matter; they will be bit-cast to u32.
   // TODO: Add $convertU32PairToI53Signaling() variant.
   $convertU32PairToI53: function(lo, hi) {
-    return (lo >>> 0) + (hi >>> 0) * 4294967296;
-  }
+    var result = (lo >>> 0) + (hi >>> 0) * 4294967296;
+#if ASSERTIONS
+    if ((result >>> 0) != (lo >>> 0)) warnOnce('convertU32PairToI53() resulted in rounding (lo=0x' + lo + ', hi=0x' + hi + ') to 0x' + result.toString(16));
```

Comment:
Why wouldn't I want to warn each time this happens?   Also, why not make it an assertion?  

If there valid use cases for passing out-of-range values to this API, then presumably the developer will want to way to disable the warning or assertion whichever one we go with.   

If there are no valid reasons then I think assertion makes more sense.